packages/stark-core/src/common/bootstrap/abstract-stark-main.ts

```diff
@@ -0,0 +1,163 @@
+import { ApplicationRef, ComponentRef, enableProdMode, NgModuleRef } from "@angular/core";
+import { disableDebugTools, enableDebugTools } from "@angular/platform-browser";
+import { createNewHosts } from "@angularclass/hmr";
+
+import { StarkMain } from "./stark-main.intf";
+import {StarkEnvironment} from "../environment";
+
+/**
+ * Parent class for bootstrapping Stark applications.
+ * It handles the DOMContentLoaded event of the Web browser in order to bootstrap the application.
+ * This class also supports HMR in development mode.
+ */
+export abstract class AbstractStarkMain implements StarkMain {
+	protected constructor(protected environment: StarkEnvironment) {
+		// no-op
+	}
+
+	/**
+	 * Must be implemented by children classes. Called within bootstrapDomReady (i.e., when the DOM has been loaded)
+	 */
+	public abstract main: () => Promise<any>;
+
+	public mainWrapper = (): void => {
```

Comment:
You use different ways to define methods in this file, I think keep only one.
Could we have
> public mainWrapper(): void { ... }

instead of
> public mainWrapper = (): void => { ... }

Reply:
I changed to a lambda to capture "this", otherwise all calls to this would fail because they're made through a callback

Reply:
But I have in mind that there was another way to do it, just can't remember it though :p